docusaurus-search-local/src/server/utils/processDocInfos.ts

```diff
@@ -16,76 +17,108 @@ export function processDocInfos(
+    const versionPaths = routesPaths
+      .map<DocInfoWithRoute | undefined>((url: string) => {
+        // istanbul ignore next
+        if (!url.startsWith(baseUrl)) {
+          throw new Error(
+            `The route must start with the baseUrl "${baseUrl}", but was "${url}". This is a bug, please report it.`
+          );
+        }
+        const route = url.substr(baseUrl.length).replace(/\/$/, "");
 
-      // ignore files
-      if (
-        ignoreFiles?.some((reg: RegExp | string) => {
-          if (typeof reg === "string") {
-            return route === reg;
-          }
-          return route.match(reg);
-        })
-      ) {
-        return;
-      }
+        // Do not index homepage, error page and search page.
+        if (
+          ((!docsRouteBasePath || docsRouteBasePath[0] !== "") && route === "") ||
+          route === "404.html" ||
+          route === "search"
+        ) {
+          return;
+        }
 
-      if (
-        indexBlog &&
-        blogRouteBasePath.some((basePath) => isSameOrSubRoute(route, basePath))
-      ) {
+        // ignore files
         if (
-          blogRouteBasePath.some(
-            (basePath) =>
-              isSameRoute(route, basePath) ||
-              isSameOrSubRoute(route, path.posix.join(basePath, "tags"))
-          )
+          ignoreFiles?.some((reg: RegExp | string) => {
+            if (typeof reg === "string") {
+              return route === reg;
+            }
+            return route.match(reg);
+          })
         ) {
-          // Do not index list of blog posts and tags filter pages
           return;
         }
-        return { route, url, type: "blog" };
-      }
-      if (
-        indexDocs &&
-        docsRouteBasePath.some((basePath) => isSameOrSubRoute(route, basePath))
-      ) {
-        return { route, url, type: "docs" };
-      }
-      if (indexPages) {
-        return { route, url, type: "page" };
-      }
-      return;
-    })
-    .filter<DocInfoWithRoute>(Boolean as any)
-    .map(({ route, url, type }) => ({
-      filePath: path.join(
-        outDir,
-        siteConfig.trailingSlash === false
-          ? `${route}.html`
-          : `${route}/index.html`
-      ),
-      url,
-      type,
-    }));
+
+        if (
+          indexBlog &&
+          blogRouteBasePath.some((basePath) => isSameOrSubRoute(route, basePath))
+        ) {
+          if (
+            blogRouteBasePath.some(
+              (basePath) =>
+                isSameRoute(route, basePath) ||
+                isSameOrSubRoute(route, path.posix.join(basePath, "tags"))
+            )
+          ) {
+            // Do not index list of blog posts and tags filter pages
+            return;
+          }
+          return { route, url, type: "blog" };
+        }
+        if (
+          indexDocs &&
+          docsRouteBasePath.some((basePath) => isSameOrSubRoute(route, basePath))
+        ) {
+          if (docs.size === 0 || docs.has(url)) {
```

Comment:
The diff is a bit messy here but this is basically the only change I made in the map function.

Reply:
Yes I can see it by enabling "hide whitespace" in diff view settings